astropy/convolution/utils.py

```diff
@@ -86,7 +86,46 @@ def add_kernel_arrays_2D(array_1, array_2):
     return array_2 + array_1
 
 
-def discretize_model(model, x_range, y_range=None, mode="center", factor=10):
+def add_kernel_arrays_3D(array_1, array_2):
+    """
+    Add two 3D kernel arrays of different size.
+
+    The arrays are added with the centers lying upon each other.
+    """
+    if array_1.size > array_2.size:
```

Comment:
Can the duplicate code here be removed by just doing `if array_1.size < array_2.size: array_1, array_2 = array_2, array_1`?  I don't see any reason it can't

Reply:
there are a lot of opportunities to reduce duplication indeed. This one especially could be applied to 1D and 2D code too. I'm hesitant to do it right now because I have a slight preference to keep the code as similar to the 2D version as possible at first. Is it ok to postpone deduplication refactors to when the scope of the PR is better defined ?